### PR TITLE
:bug: Fix password complexity description generation.

### PR DESCRIPTION
**Pull request by alexeystadnikov-okta**
Resolves OKTA-113809

screen shot 2017-02-10 at 4 02 25 pm
screen shot 2017-02-10 at 4 06 23 pm

@magizh-okta 
@mauriciocastillosilva-okta 
@rchild-okta 

Please, review.

### CODE REVIEW

**Comment by mauriciocastillosilva-okta** [comment]
@edjohns-okta in case you are interested

File: test/unit/helpers/xhr/PASSWORD_RESET_withMinComplexity.js [hunk 1]
```diff
@@ -8,7 +8,12 @@ define({
     "_embedded": {
       "policy": {
         "complexity": {
-          "minLength": 8
```
Comment (alexeystadnikov-okta) [inline test/unit/helpers/xhr/PASSWORD_RESET_withMinComplexity.js]: Done

File: test/unit/spec/PasswordReset_spec.js [hunk 2]
```diff
@@ -12,31 +13,43 @@ define([
...
+  }
+
   function setup(settings) {
     var passwordResetResponse = resPasswordReset;
-    if(settings) {
-      switch (settings.policyComplexity) {
-      case 'min':
-        passwordResetResponse = resPasswordResetWithMinComplexity;
-        break;
+    var policyMaxComplexityDefaults = {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/PasswordReset_spec.js]: nit: can we name this ```policyComplexityDefaults```? It contains all properties, but not really a max of any value.

File: test/unit/spec/PasswordReset_spec.js [hunk 3]
```diff
@@ -12,31 +13,43 @@ define([
...
+
+    if (settings && settings.policyComplexity) {
+      passwordResetResponse = deepClone(resPasswordResetWithComplexity);
+      var responsePolicy = passwordResetResponse.response._embedded.policy;
 
-      case 'max':
-        passwordResetResponse = resPasswordResetWithMaxComplexity;
-        break;
+      var key = settings.policyComplexity;
+      if (key === 'max') {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/PasswordReset_spec.js]: nit: following the same nit on my other comment, can this be ```key === 'all'``` or ```key === 'full'```?

File: test/unit/spec/PasswordReset_spec.js [hunk 4]
```diff
@@ -100,12 +113,41 @@ function (Q, _, $, OktaAuth, LoginUtil, Util, PasswordResetForm, Beacon, Expect,
...
+        expect(test.form.subtitleText()).toEqual('Your password must have a symbol.');
+      });
+    });
+    itp('has a valid subtitle if only password complexity "excludeUsername" defined', function () {
+      return setup({policyComplexity: 'excludeUsername'}).then(function (test) {
+        expect(test.form.subtitleText()).toEqual('Your password must have no parts of your username.');
+      });
+    });
+
     itp('has a valid subtitle if MAX password complexity defined', function () {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/PasswordReset_spec.js]: nit: same as the other nits, the description here can be ```has a valid subtitle if password complexity is defined with all possible options```

File: test/unit/spec/PasswordReset_spec.js (hunk 2 above)
Comment (alexeystadnikov-okta) [inline test/unit/spec/PasswordReset_spec.js]: Done

File: test/unit/spec/PasswordReset_spec.js (hunk 3 above)
Comment (alexeystadnikov-okta) [inline test/unit/spec/PasswordReset_spec.js]: Done

File: test/unit/spec/PasswordReset_spec.js (hunk 4 above)
Comment (alexeystadnikov-okta) [inline test/unit/spec/PasswordReset_spec.js]: Done

**Comment by alexeystadnikov-okta** [comment]
Squashed and rebased.

**Pull request merged**